tests/models/opt/test_modeling_opt.py

```diff
@@ -63,7 +63,7 @@ def __init__(
         use_labels=False,
         vocab_size=99,
         hidden_size=16,
-        num_hidden_layers=2,
+        num_hidden_layers=5,
```

Comment:
Good for me for the change here!

Without going through the work about `model parallelism and offload` so far, I am wondering if the method `dispatch_model` (called in `from_pretrained`) would fail if a user has a tiny model.

I know, this is very unlikely.

Reply:
It does require a model of a size bigger than a couple of bytes otherwise you hit some edge cases.

Reply:
Thanks @sgugger . I would like to click the approval button if I could understand better why increasing `max_size` values will make work for T5 🙏

Reply:
I don't understand what argument you are talking about, there is no `max_size` in any of the functions/methods. Are you talking about `max_memory`? It sets the maximum memory available on each device.

Reply:
I  should be more precise, sorry.

There is a change from
```
max_size = int(0.4 * model_size)
```
to
https://github.com/huggingface/transformers/blob/05902a36eda246b874d91bef3518e0eab8026ade/tests/test_modeling_common.py#L2233

And for T5 test, you set `model_split_percent`

https://github.com/huggingface/transformers/blob/05902a36eda246b874d91bef3518e0eab8026ade/tests/models/t5/test_modeling_t5.py#L518-L519.

Despite the comment `# The small T5 model needs higher percentages for CPU/MP tests`, I still couldn't understand why we need larger values in `model_split_percent` (which is used to get `max_size = int(self.model_split_percents[0] * model_size) `).

- I understand we need models to have a minimal size as you explained
- but why `maximum memory available on each device` also plays  a role in these tests.

Reply:
The maximum size needs to be able to accommodate the largest layer of the model plus the layers that are on it (for GPU and CPU RAM). Since this model is so tiny (2 layers for the encoder and decoder) with some shared weights, it needs to be switched to something higher than half the size of the model or it doesn't work.

This is just a edge case because of tiny models and tiny maximum sizes that does not appear in real life, but I can't test on large models :-)